### PR TITLE
Improve ForEnvironment to not overwrite config data

### LINKED ISSUE #73: Cannot use 'environment' as a key in the configuration json
It was not clear to me that I could not use the word "environment" as a key in my configuration json. It might be worth including that in the documentation.

I'm not sure if this is the case for ruby or python, but it certainly is for C#. The way it manifests itself is by completely ignoring sections named "environment"

### PR DESCRIPTION
**Pull request by jtroe**
Fixes #73

If this looks like a good solution, I'll implement in Ruby and Python.

### CODE REVIEW

**Comment by pdenny** [comment]
Also, I don't think that this approach will solve the problem. Which is that when a container's key is "environment", then it is ignored. My environment container (now named env) is a pretty big block within each greater environment section, so everything inside it (which is an entire hierarchy in and of itself) is left as it was in the 'all' section. But there is a very real possibility that I don't understand what is going on behind the scenes well enough to see what your code should be doing.

**Comment by jtroe** [comment]
Fixed it so it builds and tests properly.

Added a snippet in the doc to show what I'm trying to accomplish.

**Comment by pdenny** [comment]
Looks like it works swimmingly both via the unit tests and running it in my bpxa solution. I can name my env block either 'env' or 'environment' and it works great. Thanks!

**Comment by pdenny** [comment]
Time for release version 1.2.1.0?

**Comment by jtroe** [comment]
I'll add the changes to Python and Ruby and we can kick another release out.

Unless you want to take a stab at the changes, @pdenny 😁

**Comment by pdenny** [comment]
Still trying to get this BPXA bit of work out the door. If you can't get to it, I can take care of this next week.

File: ruby/lib/centroid.rb [hunk 1]
```diff
@@ -59,14 +59,19 @@ def to_s
...
       if all_key.nil?
-        Config.new(env_json)
+        config = Config.new(env_json)
       else
         all_json = raw_config[all_key]
-        Config.new(deep_merge(all_json, env_json))
+        config = Config.new(deep_merge(all_json, env_json))
       end
+
+      if !config.has_key?('environment')
```
Comment (jtroe) [inline ruby/lib/centroid.rb]: Alternately:
```
return config unless !config.has_key?('environment')
config.raw_config['environment'] = env
config
```

**Comment by pdenny** [comment]
Changes look good. All tests pass on my system and I was able to use the ruby gem here at BPXA. I didn't have any external projects to double check the python updates with. :shipit:  and kick out a release!

**Pull request merged**